File: fib.rb

def fib(num)
	if num==0 then
		return 0
	end

	if num==1 then
		return 1
	end

	return fib(num-1) + fib(num-2)
end

n=0

if ARGV[0] then
	n=ARGV[0].to_i
end

result=fib(n)
puts "result: #{result}"

File: fib.rs

fn fib(n: i32) -> i32{
	if n==0 {
		return 0;
	}
	if n==1 {
		return 1;
	}
	return fib(n-1)+fib(n-2);
}

fn main(){
	let n = 44;
	let result = fib(n);
	println!("result of {} is:{}",n,result);
}
